x/bank/keeper/view.go

```diff
@@ -67,6 +68,32 @@ func (k BaseViewKeeper) GetAllBalances(ctx sdk.Context, addr sdk.AccAddress) sdk
 	return balances.Sort()
 }
 
+// GetAccountsBalances returns all the accounts balances from the store.
+func (k BaseViewKeeper) GetAccountsBalances(ctx sdk.Context) []types.Balance {
+	balances := make([]types.Balance, 0)
+	mapAddressToBalancesIdx := make(map[string]int)
+
+	k.IterateAllBalances(ctx, func(addr sdk.AccAddress, balance sdk.Coin) bool {
```

Comment:
Is IterateAllBalances deterministic? i.e. If I run `GetAccountsBalances` twice, will the same address be at the same index in `balances` array?

I believe @sahith-narahari's original idea was to 1/ sort the coins in each balance (this is done here), and 2/ sort the `balances` array by address

Reply:
> Is IterateAllBalances deterministic? i.e. If I run GetAccountsBalances twice, will the same address be at the same index in balances array?

yes it's deterministic. The reason why we had to sort the addresses was that we were iterating over the map items, which is non-deterministic. Here we only use the map for lookup, so we don't need to sort the balances by address 👍